Automated response actions for rules (#3546)
elastic/security-docs: docs/osquery/osquery-response-action.asciidoc

== Add Osquery Response Actions
preview::[]

> :frontmatter-description: Osquery Response Actions allow you to add live queries to custom query rules so you can automatically collect data on systems the rule is monitoring. 

Reviewer: ```suggestion
:frontmatter-description: Osquery Response Actions allow you to add live queries to custom query rules so you can automatically collect data on systems the rules are monitoring. 
```

[[find-osquery-response-action-results]]
=== Find query results

> When an alert is generated, Osquery automatically collects data on the system related to the alert. Query results are displayed within the *Response Results* tab in the Alert details flyout. The number next to the *Response Results* tab represents the number of queries attached to the rule, in addition to endpoint response actions ran by the rule.

Reviewer: ```suggestion
When a host generates an alert, Osquery automatically collects data about that host. Query results are displayed within the *Response Results* tab in the Alert details flyout. The number next to the *Response Results* tab represents the number of queries attached to the rule, in addition to endpoint response actions run by the rule.
```
Could be wrong here, but if the first sentence is accurate like this I think it's more clear (active voice too). Seems like "run" not "ran" since its not strictly about rules that ran in the past?

Reviewer 2: I also like @benironside's active voice for the first sentence, but it's not quite accurate to say that a host generates an alert — a rule generates an alert, using data from the host. Maybe:
```suggestion
When a rule generates an alert, Osquery automatically collects data about the host involved. Query results are displayed within the *Response Results* tab in the Alert details flyout. The number next to the *Response Results* tab represents the number of queries attached to the rule, in addition to endpoint response actions run by the rule.
```
(I'm not 100% sold on "the host involved," could also be "the associated host" or "the affected host" or similar?)

And I'd go with `run`, because `ran` is just plain past tense but this sentence needs the past participle `run`, as in "actions _that were_ run by the rule" or "actions _that had been_ run by the rule."

Author: I think we can make it more direct and just say that the Osquery action collects data on the host:

```suggestion
When an alert is generated, Osquery automatically collects data on the host. Query results are displayed within the *Response Results* tab in the Alert details flyout. The number next to the *Response Results* tab represents the number of queries attached to the rule, in addition to endpoint response actions ran by the rule.
```
